nikola/nikola.py

```diff
@@ -2404,8 +2404,11 @@ def generic_index_renderer(self, lang, posts, indexes_title, template_name, cont
             while posts:
                 lists.append(posts[:kw["index_display_post_count"]])
                 posts = posts[kw["index_display_post_count"]:]
+            if not lists:
+                lists.append([])
         num_pages = len(lists)
-        displayed_page_numbers = [utils.get_displayed_page_number(i, num_pages, self) for i in range(num_pages or 1)]
+        assert num_pages >= 1
```

Comment:
Assertions are ugly and can be disabled, use an `if` and raise an exception yourself.